### DISCUSSION TITLE
Is Cropping Video Best Practice?

### DISCUSSION THREAD

**Discussion by jmdelahanty** [talmolab/sleap#804] [General]
Hello sleap-y heads!

I was wondering if cropping videos before feeding them into SLEAP for analysis is helpful/useful for the algorithm. 

The edges of the video's frame are locations that the mouse (shouldn't) ever end up in by design. Since the mouse won't be there, I'm wondering if it's better practice for videos to crop them.

I know that the network doesn't need the help/wouldn't likely label the mouse in those spots outside the frame since the training data won't have examples of it, so intuitively I would imagine it doesn't matter/doesn't help at all. It also probably doesn't make the video files much smaller/much easier to fit into the GPU's memory.

But still, I was just thinking about whether it's a general best practice to do so.

So is it helpful/better to crop your videos in general? Or does it not matter?

**Comment by talmo** [comment]
If it'll significantly reduce the frame size, it might be worth it, but most of the time it's not since it takes a good amount of processing to read, crop, and re-encode the videos. Ideally you'll set that up on the camera/acquisition side.

Also, if you're using a top down model this will be even more negligible since only the first stage will process the entire image (and at low resolution).
